### DISCUSSION TITLE
CRP name changeability?

### DISCUSSION THREAD

**Discussion by Cryptic1526** [maximmax42/Discord-CustomRP#213] [Ideas]
I've used CRP for a long time now, but I;ve recently seen a Vencord CLI Plugin that does the same thing CRP does, with the added feature of changing the names of the activity, without needing to add the name directly to an app in discord dev portal. The only drawback is that it's not quickly accessable and it doesn't have a easy to use interface and preset feature like in CRP, and I think adding a box to set the name of the RP would be nice.

**Comment by maximmax42** [comment]
I have no idea how they are able to change app's name, I'll look into it.

**Reply by Cryptic1526** [reply]
I know Vendicated, I'll try asking him for some source code, I'll lyk how it goes (I might also be able to get some info about adding different options for "Playing...", like "Listening to..." & "Watching..."). Don't keep your hopes up tho, it's like talking to a wall when you ask them something 💀

**Reply by Cryptic1526** [reply]
@maximmax42 
![image](https://github.com/maximmax42/Discord-CustomRP/assets/119151348/97131327-9e88-42c5-8d12-be6bb8912756)

https://github.com/Vendicated/Vencord/blob/main/src/plugins/customRPC.tsx#L77-L82
https://github.com/Vendicated/Vencord/blob/main/src/plugins/customRPC.tsx#L160

See if that helps?

**Comment by maximmax42** [comment]
> @maximmax42 ![image](https://user-images.githubusercontent.com/119151348/256477297-97131327-9e88-42c5-8d12-be6bb8912756.png)
> 
> https://github.com/Vendicated/Vencord/blob/main/src/plugins/customRPC.tsx#L77-L82 https://github.com/Vendicated/Vencord/blob/main/src/plugins/customRPC.tsx#L160
> 
> See if that helps?

The types only work if you connect with a user's token, my app doesn't do that, so only "playing" is available. Name changing might also be only available with a token.

**Comment by Cryptic1526** [comment]
> The types only work if you connect with a user's token, my app doesn't do that, so only "playing" is available. Name changing might also be only available with a token.

Hmm. I asked him and he said: "i know nothing about rich presence using the actual presence server rather than using discord internals, sorry. if i had to guess, it is the same".

So I'm at a loss there, I'll dig around my old projects and see if I did something like this before.
